### ISSUE TITLE
symlinked conf.d file loading and naming can clash with runlevel specific conf.d

### ISSUE THREAD

**Issue by gyakovlev** [OpenRC/openrc#286]
https://github.com/OpenRC/openrc/blob/3eef6e91274f2e07bd566f206e89d9b0b9c45fb9/sh/openrc-run.sh.in#L223-L237

consider user having a file `/etc/init.d/app`
and a symlink  to above file ^ `/etc/init.d/app.default`

and it's added to runlevel `default`

openrc will try to source runlevel-specific config first, which will be `/etc/conf.d/app.default.default` and on next try will try to source `/etc/conf.d/app` because bash string regex is greedy ` _c=${RC_SVCNAME%%.*} `

so basically there need to be some logic handling this case. 

or regex not as greedy (single `%`), but not sure how it will affect existing services or symlinks.

as I understand intention is to load specific file first and fallback to generic one if it fails.

but there is funky behavior if  user creates a postfix that clashes with runlevel name.

**Comment by gyakovlev** [comment]
reported by user gmturner on irc
